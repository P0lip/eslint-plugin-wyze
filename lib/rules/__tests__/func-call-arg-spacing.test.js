import { RuleTester } from 'eslint'
import rule from '../func-call-arg-spacing'

const parserOptions = {
  ecmaVersion: 2017,
  sourceType: 'module',
}

const makeMissingError = location =>
  `There should be no space ${location} '${location === 'after' ? '(' : ')'}'.`

describe('func-call-arg-spacing', () => {
  it('passes RuleTester', () => {
    new RuleTester().run('func-call-arg-spacing', rule, {
      valid: [
        {
          code:
          `
          const f = ( a, b ) => a + b

          f(1, 2)
          `,
          parserOptions,
        },
        {
          code:
          `
          const f = ( a, b ) => a + b

          f()
          `,
          parserOptions,
        },
        {
          code:
          `
          const f = ( a, b ) => a + b

          f(
            1,
            2,
          )
          `,
          parserOptions,
        },
        {
          code:
          `
          const app = { use: () => {} }

          app.use(( req, res ) => {})
          `,
          parserOptions,
        },
        {
          code: '[].filter(x => x)',
          parserOptions,
        },
      ],
      invalid: [
        {
          code:
          `
          const f = ( a, b ) => a + b

          f( 1, 2 )
          `,
          parserOptions,
          errors: [
            makeMissingError('after'),
            makeMissingError('before'),
          ],
        },
        {
          code:
          `
          const f = () => {}

          f( )
          `,
          errors: [
            makeMissingError('after'),
          ],
          parserOptions,
        },
        {
          code:
          `
          const keys = properties
            .filter( prop => !prop.computed )
            .map( prop => prop.key.name )
          `,
          errors: [
            makeMissingError('after'),
            makeMissingError('before'),
            makeMissingError('after'),
            makeMissingError('before'),
          ],
          parserOptions,
        },
      ],
    })
  })
})
