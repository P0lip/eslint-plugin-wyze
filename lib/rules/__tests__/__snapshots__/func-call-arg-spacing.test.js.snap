// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`func-call-arg-spacing applies fixes to callbacks 1`] = `
Object {
  "errorCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "/Users/neil/Projects/personal/eslint-plugin-wyze/lib/rules/__tests__/fixtures/func-call-arg-spacing/callbacks.js",
      "messages": Array [],
      "output": "
const a = properties
  .filter(prop => !prop.computed)

const b = properties
  .map(prop => prop.key.name)

const c = properties
  .filter(prop => !prop.computed)
  .map(prop => prop.key.name)

const app = { use: () => {} }

app.use(( req, res ) => {})

[ 1 ].filter(x => x)
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;

exports[`func-call-arg-spacing applies fixes to functions with no args 1`] = `
Object {
  "errorCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "/Users/neil/Projects/personal/eslint-plugin-wyze/lib/rules/__tests__/fixtures/func-call-arg-spacing/no-args.js",
      "messages": Array [],
      "output": "
const f = ( a, b ) => a + b

f()
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;

exports[`func-call-arg-spacing applies fixes to multiline functions 1`] = `
Object {
  "errorCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "/Users/neil/Projects/personal/eslint-plugin-wyze/lib/rules/__tests__/fixtures/func-call-arg-spacing/multi-line.js",
      "messages": Array [],
      "output": "
const f = ( a, b ) => a + b

f(1,
  2)
f(1,
  2)
f(1,
  2)
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;

exports[`func-call-arg-spacing applies fixes to single line functions 1`] = `
Object {
  "errorCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "/Users/neil/Projects/personal/eslint-plugin-wyze/lib/rules/__tests__/fixtures/func-call-arg-spacing/single-line.js",
      "messages": Array [],
      "output": "
const f = ( a, b ) => a + b

f(1, 2)
f(1, 2)
f(1, 2)
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;
